### PR TITLE
Fortran implementation of SMIOLf_put/get_var

### PR DESCRIPTION
**Pull request by MiCurry**
These commits contain an implementation for the put/get var routines for smiolf.F90. Currently putting and getting a character variable does not preform as expected.

### CODE REVIEW

File: src/gen_put_get_var_interfaces.sh [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+        printf "       ! \n"
+        printf "       ! Convert character variable string that is to be %s\n" $direction
+        printf "       ! NOTE: This operation is only works upon a single string\n"
+        printf "       ! \n"
+        printf "       allocate(string(1))\n"
+        printf "       allocate(string(1) %% str(len(buf)))\n"
+        printf "\n"
+        printf "       do i=1,len(buf)\n"
+        printf "           string(1) %% str(i) = buf(i:i)\n"
+        printf "       enddo\n"
```
Comment (MiCurry) [inline src/gen_put_get_var_interfaces.sh:199]: The string should only be copied if the `direction == put`.

**Comment by MiCurry** [comment]
@mgduda is this PR ready to be reviewed? As this is my PR, I don't think you can request a review from me, but I'm happy to give it another look over when its ready.

**Comment by mgduda** [comment]
@MiCurry Not yet. I just pushed a rebased branch that should be compatible with the current `master` branch. At a minimum, though, I'd like to rework the Fortran put/get interfaces so that we don't need to pass all dimension sizes. I recognize that this was originally done on account of restrictions in Fortran 2003 (and that were relaxed in Fortran 2008). One option might be to write our own functions for getting the `C_LOC` of assumed-shape arrays:
```fortran
    subroutine set_ptr3(c_buf, buf, d1, d2, d3)                                                                                      
                                                                                                                                     
        use iso_c_binding, only : c_loc, c_double, c_ptr                                                                             
                                                                                                                                     
        implicit none                                                                                                                
                                                                                                                                     
        integer :: d1, d2, d3                                                                                                        
        real(kind=c_double), dimension(d1, d2, d3), target :: buf                                                                    
        type (c_ptr) :: c_buf                                                                                                        
                                                                                                                                     
        c_buf = c_loc(buf)                                                                                                           
                                                                                                                                     
    end subroutine set_ptr3
```
and to call this routine from within, e.g., the `SMIOLf_put_var_3d_real64` routine as
```fortran
call set_ptr3(c_buf, buf, size(buf,1), size(buf,2), size(buf,3))
```
where we were previously just using `C_LOC` directly on an explicit-size array.

This would simplify the SMIOL Fortran API, especially for higher-dimensional arrays.

**Comment by mgduda** [comment]
This PR has been replaced by PR #65 .

**Pull request closed without merging**